conans/client/graph/range_resolver.py

```diff
@@ -31,7 +32,7 @@ def resolve(self, require, base_conanref, remotes, update):
         search_ref = RecipeReference(ref.name, "*", ref.user, ref.channel)
 
         resolved_ref = self._resolve_local(search_ref, version_range)
-        if resolved_ref is None or update:
+        if resolved_ref is None or should_update_reference(search_ref, update):
```

Comment:
I think that ``--update=pkg/1.1`` might still update if there is an arbitrary version range like ``pkg/[>=2]`` too, maybe it is good to cover this case in tests? Maybe update patterns should only be package names, not references?